### ISSUE TITLE
YAML inventory format should support 'children' groups like INI format can

### ISSUE THREAD

**Issue by dcoutu** [ansible/ansible#150]
Allow for nesting of groups within groups in the /etc/ansible/hosts Inventory File.  A comparative (but not exact match) feature would be LUMPs in dsh.

Examples would be something like:

[datacenters]
east-coast-dc
west-coast-dc

[east-coast-dc]
eastcoast-webservers
eastcoast-appservers

[west-coast-dc]
westcoast-webservers
westcoast-appservers

[east-coast-webservers]
web1
web2

[west-coast-webservers]
web3
web4

[site-webservers]
east-coast-webservers
west-coast-webservers

There may be some better ways to accomplish some of the nesting via something like tagging classes/roles/location of host defs, but I haven't thought that completely through yet.  Could be of the form:

host: location:class/role, i.e

web1:east-coast:webserver 
app3:west-coast:appserver
webapp2:east-coast:appserver,webserver

Need to balance extensibility against simplicity here, but limiting the amount of duplication in the configuration will help out tremendously for inheritance and reduce errors and mistakes in forgetting to add hosts to multiple groups.

**Comment by mpdehaan** [comment]
nice idea

ticket #146 may be able to make use of this in the YAML syntax (we're still supporting the old, but offering more advanced options that are hard to represent for people that want more).

We might not end up supporting this in the INI style file.

**Comment by mpdehaan** [comment]
I have implemented this for the INI format file as:

[groupname:children]
group
othergroup

and also at the same time, I have implemented:

[groupname:vars]
var1=x
var2=y

So the INI format can now do more than the YAML one!

I have not upgraded the YAML one to support child groups yet.  This could probably be done as 'children' as a list of group names underneath a group element.

The API is already there (add_child_group)
